### ISSUE TITLE
OCI: no version range support for dependencies/`helm install`

### ISSUE THREAD

**Issue by TZdybel** [helm/helm#9694]
I'm trying to make my helm chart with dependencies to get newest version I pass to it.
I see that "latest" tag is not possible so I've tried with semantic versions like "~1.x" which should work according to documentation.
It works for things like mongodb chart from bitnami repo, but not for my Azure Container Registry.
My `Chart.yaml`:
```yaml
apiVersion: v2
name: somename
description: A Helm chart for Kubernetes
version: 0.1.0
appVersion: "1.0"
dependencies:
  - name: common
    version: 1.0.0
    repository: "oci://myazurecr.azurecr.io/helm"
```
Output from `helm dependency update`:
```
Getting updates for unmanaged Helm repositories...
...Unable to get an update from the "oci://myazurecr.azurecr.io/helm" chart repository:
        tag explicitly required
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "kured" chart repository
...Successfully got an update from the "ingress-nginx" chart repository
...Successfully got an update from the "gitlab" chart repository
...Successfully got an update from the "stable" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 1 charts
Downloading common from repo oci://myazurecr.azurecr.io/helm
~1.x: Pulling from myazurecr.azurecr.io/helm/common
Save error occurred:  could not download oci://myazurecr.azurecr.io/helm/common: myazurecr.azurecr.io/helm/common:~1.x: not found
Deleting newly downloaded charts, restoring pre-update state
Error: could not download oci://myazurecr.azurecr.io/helm/common: myazurecr.azurecr.io/helm/common:~1.x: not found
```
Is this a bug related to recent introduction of OCI references as repository?

**Comment by bacongobbler** [comment]
As far as I can tell, version range support has not yet been implemented with the OCI feature. It looks like it can only lookup the literal version number.

https://github.com/helm/helm/blob/d4cc130fa90027e8c1c69ad2a06c6dee4c31d29d/pkg/getter/ocigetter.go#L43-L45

cc @jdolitsky

**Comment by cndoit18** [comment]
I will try to fix it this weekend.

**Comment by bacongobbler** [comment]
This is not something that can be fixed over the weekend, @cndoit18. The API lacks certain features preventing this functionality. More specifically, the API does not currently have a way to determine which tags are available when given a particular repository.

We need to plan and work through the design criteria before we implementing something.

**Comment by cndoit18** [comment]
> This is not something that can be fixed over the weekend, @cndoit18. The API lacks certain features preventing this functionality. More specifically, the API does not currently have a way to determine which tags are available when given a particular repository.
> 
> We need to plan and work through the design criteria before we implementing something.

wow, thank you very much

**Comment by bacongobbler** [comment]
updated the title to reflect the ask.

**Comment by sidewinder12s** [comment]
@bacongobbler So Helm cannot track an OCI repository floating chart version for a chart or it's dependencies currently?

(we use ArgoCD to manage helm charts that just float on a major version that we constantly push)

**Comment by ching08** [comment]
Hi, is there any chance that this feature will  be included  in the next releases 3.8 ( jan 2022)  ?

**Comment by scottrigby** [comment]
@sabre1041 is working on the required API on the ORAS side. Will link issues/PRs

Collaborating with folks on the Helm side as well, so we can meet in the middle.
